code-conventions.md

```diff
@@ -314,3 +314,28 @@ The exceptions are `Material.AIR`, which is a good way to represent "nothing"
 and `Material.STONE` which can be used to get a dummy `ItemMeta`.
 
 Prefer to avoid referencing the Biome enum directly, since it has changed between versons in the past.
+
+### Deprecating
+
+When deprecating a Java element (such as a class, method, or constructor), follow these conventions:
+
+1. **Replace all internal usages** of the deprecated element within the Skript codebase, where feasible.
+2. Annotate the element with `@Deprecated(forRemoval = true)` to indicate that it is scheduled for removal in a future release.
```

Comment:
`@Deprecated` should still be available for things that shouldn't be used but aren't likely to go away anytime soon imo

Reply:
I think this tends to lead to code just sticking around for no real reason
I think if you're deprecating something you should make a plan to remove it.
Exceptions can be made if needed, of course.